packages/effect/src/String.ts

```diff
@@ -747,3 +747,93 @@ const isLineBreak = (char: string): boolean => {
 const isLineBreak2 = (char0: string, char1: string): boolean => char0.charCodeAt(0) === CR && char1.charCodeAt(0) === LF
 
 const linesSeparated = (self: string, stripped: boolean): LinesIterator => new LinesIterator(self, stripped)
+
+/**
+ * Truncates a string to a specified length and by default adds the omission string "..." to the end of the string.
+ * @since 3.12.0
+ *
+ * @example
+ * import { truncate } from "effect/String";
+ * truncate("Hello World!", 5); // "Hello..."
+ */
+export const truncate: {
+  (
+    str: string,
+    lengthOrOptions:
```

Comment:
I think we should just have an options object instead of a union. Also the member should be `readonly`.